Wikipedia/Code/TalkPageContainerViewController.swift

```diff
@@ -191,6 +192,121 @@ private extension TalkPageContainerViewController {
             return fallbackGenericString
         }
     }
+    
+    func absoluteURL(for url: URL) -> URL? {
+        
+        var absoluteUrl: URL?
+        
+        if let firstPathComponent = url.pathComponents.first,
+            firstPathComponent == ".",
+            url.host == nil,
+            url.scheme == nil,
+            let siteURLHost = siteURL.host,
+            let siteURLScheme = siteURL.scheme {
+            absoluteUrl = URL(string: "\(siteURLScheme)://\(siteURLHost)/wiki/")
```

Comment:
There's a `wmf_URL(withPath:` extension on URL that might be of use here. If that doesn't fit, is there a reason not to use `URLComponents` rather than assembling the string manually?

Reply:
@joewalsh nope no reason, either of those sound better, thanks!

Reply:
@joewalsh used wmf_URL so feels a little better but cleaner might be to pass along a baseURL into wmf_attributedStringFromHTMLWithFont rather than resolving to absolute here. I explored that a little bit but ran into issues because the endpoint seems to prefix relatives with "./", checking for that and removing that path component within wmf_attributedStringFromHTMLWithFont felt risky at this point.